### PR TITLE
3206: Protect against CNI plugin being called with the host namespace

### LINKED ISSUE #3206: Protect against CNI plugin being called with the host namespace
A user had 4 Kubernetes nodes break down with this symptom:

```
$ ip route
default via 172.21.128.1 dev eth0
100.96.0.0/11 dev weave  proto kernel  scope link  src 100.101.4.0
100.96.0.0/11 dev eth0  proto kernel  scope link  src 100.101.4.194
172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.0.1
172.21.128.0/19 dev eth0  proto kernel  scope link  src 172.21.137.4
```

The `weave` and `eth0` routes are overlapping, so nothing works properly and Weave Net refuses to start.

It seems plausible that kubelet called the CNI plugin with a namespace that was actually the root namespace, so we assigned an address to `eth0` in the root.

We could check this and refuse to do it.

### PR DESCRIPTION
**Pull request by murali-reddy**
adding a check to verify if network namespace is not host network namespace

Fixes #3206

### CODE REVIEW

**Comment by bboreham** [comment]
CI failed on two counts: one lint:
> plugin/net/cni.go:127:21: error strings should not be capitalized or end with punctuation or a newline

"missing secret" is because non-weaveworks repos don't have the credentials to create VMs to run the smoke tests.

File: plugin/net/cni.go [hunk 1]
```diff
@@ -117,6 +117,16 @@ func (c *CNIPlugin) CmdAdd(args *skel.CmdArgs) error {
 	}
 	defer ns.Close()
 
+	hostNs, err := netns.Get()
+	if err != nil {
+		return fmt.Errorf("error accessing host network namespace")
```
Comment (bboreham) [inline plugin/net/cni.go]: Best to include the netns error in the error returned.

**Comment by rade** [comment]
What is happening with this?

**Comment by bboreham** [comment]
I've pushed the branch to the `weaveworks/weave` repo so it will run fill smoke tests.

**Comment by bboreham** [comment]
Thanks!

**Pull request merged**